### PR TITLE
Elitetorrent new pluging with search

### PR DESCRIPTION
**Pull request by malkavi**
Plugin to download and search in elitetorrent (spanish web)

### CODE REVIEW

File: flexget/plugins/urlrewrite_elitetorrent.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+    @plugin.internet(log)
+    def parse_download_page(self, url):
+        try:
+            page = requests.get(url).content
+            soup = get_soup(page, 'html.parser')
+            download_link = soup.findAll(href=re.compile('/get-torrent/\d+'))
+            download_href = 'http://www.elitetorrent.net' + download_link[0]['href']
+            return download_href
+        except Exception:
```
Comment (gazpachoking) [inline flexget/plugins/urlrewrite_elitetorrent.py]: We should be using more specific exceptions here.

File: flexget/plugins/urlrewrite_elitetorrent.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+    @plugin.internet(log)
+    def parse_download_page(self, url):
+        try:
+            page = requests.get(url).content
+            soup = get_soup(page, 'html.parser')
+            download_link = soup.findAll(href=re.compile('/get-torrent/\d+'))
+            download_href = 'http://www.elitetorrent.net' + download_link[0]['href']
+            return download_href
+        except IndexError:
```
Comment (stevezau) [inline flexget/plugins/urlrewrite_elitetorrent.py]: There are multiple excpetions that need to be checked..

Firstly requests exceptions. ie http errors and timeouts.. more info here http://docs.python-requests.org/en/latest/user/quickstart/#errors-and-exceptions

Second soup parse errors http://www.crummy.com/software/BeautifulSoup/bs4/doc/#errors-when-parsing-a-document

File: flexget/plugins/urlrewrite_elitetorrent.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    def url_rewrite(self, task, entry):
+        entry['url'] = self.parse_download_page(entry['url'])
+
+    @plugin.internet(log)
+    def parse_download_page(self, url):
+        try:
+            page = requests.get(url).content
+            soup = get_soup(page, 'html.parser')
+            download_link = soup.findAll(href=re.compile('/get-torrent/\d+'))
+            download_href = 'http://www.elitetorrent.net' + download_link[0]['href']
```
Comment (stevezau) [inline flexget/plugins/urlrewrite_elitetorrent.py]: Should check if downlod_link was actually found

File: flexget/plugins/urlrewrite_elitetorrent.py [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+        return (url.startswith('http://www.elitetorrent.net/torrent/'))
+
+    # urlrewriter API
+    def url_rewrite(self, task, entry):
+        entry['url'] = self.parse_download_page(entry['url'])
+
+    @plugin.internet(log)
+    def parse_download_page(self, url):
+        try:
+            page = requests.get(url).content
```
Comment (stevezau) [inline flexget/plugins/urlrewrite_elitetorrent.py]: should use tasks.requests

**Comment by liiight** [comment]
@malkavi if this is still relevant please look into the comments

**Pull request closed without merging**